internal/ethapi/api.go

```diff
@@ -1581,6 +1581,8 @@ func (s *PublicTransactionPoolAPI) GetRawTransactionByHash(ctx context.Context,
 func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, hash common.Hash) (map[string]interface{}, error) {
 	tx, blockHash, blockNumber, index, err := s.b.GetTransaction(ctx, hash)
 	if err != nil {
+		// Returns nil on purpose instead of err
+		// When the transaction doesn't exist, the return value of this RPC call should be JSON null.
```

Comment:
This method can also be invoked from the Geth console so I'm not sure mentioning the JSON encoding of a `nil` return from this function makes sense here.